qcodes_contrib_drivers/drivers/Spectrum/M4i.py

```diff
@@ -695,59 +733,152 @@ def _read_channel(self, channel, memsize=None):
+            self.external_trigger_level_1(level1)
 
     # Note: the levels need to be set in bits, not voltages! (between -8191 to
     # 8191 for 14 bits)
     def set_channel_OR_trigger_settings(self, i, trig_mode, bitlevel0, bitlevel1=None):
         """When a channel is used for triggering it must be enabled during the
         acquisition."""
         self.trigger_or_mask(0)
-        self.channel_or_mask(getattr(pyspcm, 'SPC_TMASK0_CH{}'.format(i)))
-        getattr(self, 'trigger_channel_{}_level_0'.format(i))(bitlevel0)
+        self.channel_or_mask(getattr(pyspcm, f'SPC_TMASK0_CH{i}'))
+        self.set(f'trigger_channel_{i}_level_0', bitlevel0)
         if(bitlevel1 != None):
-            getattr(self, 'trigger_channel_{}_level_1'.format(i))(bitlevel1)
-        getattr(self, 'trigger_mode_channel_{}'.format(i))(
-            trig_mode)  # trigger mode
+            self.set(f'trigger_channel_{i}_level_1', bitlevel1)
+        self.set(f'trigger_mode_channel_{i}', trig_mode)  # trigger mode
+
+    def setup_multi_recording(self, posttrigger_size, n_triggers=1,
+                              pretrigger_size=None, boxcar_average=False):
+        """ Setup multi recording.
+
+        Triggering must have been configured separately.
+        Data acquisition must started with start_triggered().
+        The collected data can be acquired with the function get_data().
+
+        Args:
+            posttrigger_size (int): size of data trace after triggering
+            n_triggers (int): total number of triggers
+            pretrigger_size (int): size of data trace before triggering
+            boxcar_average (bool): use mode 'boxcar average'
+
+        Example:
+            digitizer.setup_multi_recording(size, n_triggers)
+            digitizer.start_triggered()
+            data = digitizer.get_data()
+            digitizer.start_triggered()
+            data = digitizer.get_data()
+
+        """
+        if boxcar_average:
+            if self.oversampling_factor() != 1:
+                raise Exception('Averaging with BOXCAR can only be used with '
+                                'maximum sample rate')
+            self.card_mode(pyspcm.SPC_REC_STD_BOXCAR)
+        else:
+            self.card_mode(pyspcm.SPC_REC_STD_MULTI)
+
+        if not pretrigger_size:
+            pretrigger_size = self.pretrigger_memory_size()
+        posttrigger_size = self._hw_memsize(posttrigger_size)
+        seg_size = self._hw_memsize(posttrigger_size+pretrigger_size)
+        memsize = self._hw_memsize(n_triggers * seg_size)
+        self.data_memory_size(memsize)
+        self.segment_size(seg_size)
+        self.posttrigger_memory_size(posttrigger_size)
+
+    def start_triggered(self):
+        '''Starts triggered acquisition
+        '''
+        self.general_command(pyspcm.M2CMD_CARD_START
+                             | pyspcm.M2CMD_CARD_ENABLETRIGGER)
+
+    def get_data(self):
+        ''' Reads measurement data from the digitizer.
```

Comment:
double quote

Reply:
@FarBo, Hi Farzad, you commented on the code, but also approved the changes. Do I still have to fix the code?

Reply:
I forgot to write a final comment in the approve. Now, it is there :)